api/server/agent.go

```diff
@@ -32,21 +36,22 @@ func (tapi ToDDApi) Agent(w http.ResponseWriter, r *http.Request) {
 		// Make sure UUID string actually contains something
 		if len(uuid[0]) > 0 {
 			// Let's get the full list so we can identify the right agent if the user specified a short
-			full_agent_list := tdb.DatabasePackage.GetAgents()
+			full_agent_list, _ := tapi.tdb.GetAgents()
 
 			for i := range full_agent_list {
 				if strings.HasPrefix(full_agent_list[i].Uuid, uuid[0]) {
-					agent_list = append(agent_list, tdb.DatabasePackage.GetAgent(full_agent_list[i].Uuid))
+					agent, _ := tapi.tdb.GetAgent(full_agent_list[i].Uuid)
+					agent_list = append(agent_list, *agent)
 					break
 				}
 			}
 
 		} else { // UUID not provided; get all agents
-			agent_list = tdb.DatabasePackage.GetAgents()
+			agent_list, _ = tapi.tdb.GetAgents()
 		}
 
 	} else { // UUID not provided; get all agents
-		agent_list = tdb.DatabasePackage.GetAgents()
+		agent_list, _ = tapi.tdb.GetAgents()
```

Comment:
Thinking that you could just return an empty ```[]defs.AgentAdvert``` if there's an error here. Also should log to the server log that there was a problem.